tests/Unit/Domain/ShoppingListTest.php

```diff
@@ -54,11 +56,23 @@ public function testAddItem(): void
         $list = new ShoppingList(
             new Slug('my-groceries'),
             'My Groceries',
-            new ShoppingItemStack($item1),
+            false,
+            new ShoppingItemStack($item1)
         );
 
         $list->addItem($item2);
 
         $this->assertSame([$item1, $item2], $list->getItems()->all());
     }
+
+    /**
+     * @param ShoppingList $list
+     * @depends testConstructWithoutList
+     */
+    public function testSetArchived(ShoppingList $list): void
+    {
+        $this->assertSame($list, $list->setArchived(false));
```

Comment:
Required: the test here is not set up correctly, which is why you've not picked up on the code in `setArchived()` no longer working. You already know that a list starts off with archived as `false` - so setting it to `false` doesn't change anything.

Given: a shopping list that is not archived...
When: I set the archived flag to `true`
Then: it is archived (indicated by `isArchived()` returning `true`).